src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php

```diff
@@ -255,23 +271,41 @@ public function save($uuid, array $data)
     public function delete($uuid)
     {
         $document = $this->find($uuid);
+        $webspaceKey = $this->documentInspector->getWebspace($document);
         $this->documentManager->remove($document);
 
+        $this->documentDomainEventCollector->collect(new CustomUrlRemovedEvent(
+            $document->getUuid(),
+            $document->getTitle(),
+            $webspaceKey
+        ));
+
         return $document;
     }
 
     public function deleteRoute($webspaceKey, $uuid)
     {
         $routeDocument = $this->findRoute($uuid);
 
+        $customUrlDocument = $routeDocument->getTargetDocument();
+        while ($customUrlDocument instanceof RouteDocument) {
+            $customUrlDocument = $customUrlDocument->getTargetDocument();
+        }
```

Comment:
What is happening here where does this change come from? Seems unrelated to the PR?

Reply:
`$customUrlDocument->getTargetDocument()` might be another route in the case of a history route. will add a comment